include/nbl/asset/IRayTracingPipeline.h

```diff
@@ -0,0 +1,171 @@
+#ifndef _NBL_ASSET_I_RAY_TRACING_PIPELINE_H_INCLUDED_
+#define _NBL_ASSET_I_RAY_TRACING_PIPELINE_H_INCLUDED_
+
+#include "nbl/asset/IShader.h"
+#include "nbl/asset/IPipeline.h"
+
+#include <span>
+
+
+namespace nbl::asset
+{
+
+class IRayTracingPipelineBase : public virtual core::IReferenceCounted
+{
+  public:
+    struct SShaderGroupsParams
+    {
+      constexpr static inline uint32_t ShaderUnused = 0xffFFffFFu;
+
+      struct SGeneralShaderGroup
+      {
+        uint32_t shaderIndex = ShaderUnused;
+      };
+
+      struct SHitShaderGroup
+      {
+        uint32_t closestHitShaderIndex = ShaderUnused;
+        uint32_t anyHitShaderIndex = ShaderUnused;
+        uint32_t intersectionShaderIndex = ShaderUnused;
+      };
+
+      SGeneralShaderGroup raygenGroup;
+      std::span<SGeneralShaderGroup> missGroups;
+      std::span<SHitShaderGroup> hitGroups;
+      std::span<SGeneralShaderGroup> callableGroups;
+
+      inline uint32_t getShaderGroupCount() const
+      {
+        return 1 + hitGroups.size() + missGroups.size() + callableGroups.size();
+      }
+
+    };
+    using SGeneralShaderGroup = SShaderGroupsParams::SGeneralShaderGroup;
+    using SHitShaderGroup = SShaderGroupsParams::SHitShaderGroup;
+    struct SCachedCreationParams final
+    {
+      uint32_t maxRecursionDepth;
+    };
```

Comment:
shouldn't be nested inside `SShaderGroupsParams` just inside `IRayTracingPipelineBase` is enough